### PR TITLE
issue#325

### PR DESCRIPTION
**Pull request by adxgun**
Hi, please check the PR and see if the requirement is satisfied

### CODE REVIEW

File: p2p/service/service.go [hunk 1]
```diff
@@ -14,7 +14,7 @@ type Message interface {
 // Service is an interface that represents a networking service (ideally p2p) that we can use to send messages or listen to incoming messages
 type Service interface {
 	Start() error
-	RegisterProtocol(protocol string) chan Message
+	RegisterProtocol(protocol string, bufferSize int) chan Message
```
Comment (almogdepaz) [inline p2p/service/service.go]: i would rather you pass the chan into RegisterProtocol 
all places using RegisterProtocol should be changed accordingly

**Comment by almogdepaz** [comment]
hi @adigunhammedolalekan thanks for the pull request, looks good!
i commented on relevant parts, also notice your changes are breaking some of the tests please make sure none of the tests are failing

File: p2p/service/service.go (hunk 1 above)
Comment (adxgun) [reply p2p/service/service.go]: I didn't get that, *pass the chan into RegisterProtocol()*

Did you mean like changing RegisterProtocol signature to accept chan as parameter? 
Something like `RegisterProtocol(name, chan Message)`

**Comment by adxgun** [comment]
Thanks @almogdepaz

I'll fix that ASAP

File: p2p/service/service.go (hunk 1 above)
Comment (almogdepaz) [reply p2p/service/service.go]: yes thats right

**Comment by antonlerner** [comment]
Hi @adigunhammedolalekan 
Please fix tidy checks and re run tests so we can move along and try merge your fix

**Comment by adxgun** [comment]
Hi, @antonlerner i have been trying to fix the test issue, i will fix it as soon as possible.

Thanks

File: p2p/service/service.go (hunk 1 above)
Comment (almogdepaz) [reply p2p/service/service.go]: hi @adigunhammedolalekan are you planning to fix this ?

**Comment by adxgun** [comment]
@antonlerner @almogdepaz

File: p2p/service/service.go (hunk 1 above)
Comment (almogdepaz) [reply p2p/service/service.go]: @adigunhammedolalekan hi, nice work with the pr but you left this part out
can you fix this so we can merge ?

Comment (adxgun) [reply p2p/service/service.go]: Yes! I'll really love to, but can you please provide a little bit more details? I'll really appreciate that.

Comment (almogdepaz) [reply p2p/service/service.go]: > I didn't get that, _pass the chan into RegisterProtocol()_
> 
> Did you mean like changing RegisterProtocol signature to accept chan as parameter?
> Something like `RegisterProtocol(name, chan Message)`

you need to change  RegisterProtocol like we discussed and refactor the places it is used so the tests still pass.
so basically the only thing to do is change RegisterProtocol(protocol string, bufferSize int) chan Message to RegisterProtocol(protocol string, chan Message)
 of course that means that you will have to create and pass the chan when calling RegisterProtocol.

File: p2p/service/sim.go [hunk 2]
```diff
@@ -191,7 +192,7 @@ func (sn *Node) SubscribePeerEvents() (chan crypto.PublicKey, chan crypto.Public
 
 // RegisterProtocol creates and returns a channel for a given protocol.
 func (sn *Node) RegisterProtocol(protocol string) chan Message {
-	c := make(chan Message)
+	c := make(chan Message, config.ConfigValues.BufferSize)
```
Comment (y0sher) [inline p2p/service/sim.go]: Please use `DefaultConfig()` instead because we want to get rid of global shared state (`ConfigValues`)

Comment (almogdepaz) [inline p2p/service/sim.go:195]: you need to change RegisterProtocol like we discussed and refactor the places it is used so the tests still pass.
so basically the only thing to do is change RegisterProtocol(protocol string, bufferSize int) chan Message to RegisterProtocol(protocol string, chan Message)
of course that means that you will have to create and pass the chan when calling RegisterProtocol.

**Comment by antonlerner** [comment]
closed due to lack of activity
